nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/json/JsonPathRowRecordReader.java

```diff
@@ -0,0 +1,158 @@
+import org.apache.nifi.serialization.record.Record;
+import org.apache.nifi.serialization.record.RecordFieldType;
+import org.apache.nifi.serialization.record.RecordSchema;
+import org.apache.nifi.serialization.record.type.ArrayDataType;
+import org.apache.nifi.serialization.record.type.RecordDataType;
+import org.apache.nifi.serialization.record.util.DataTypeUtils;
+import org.apache.nifi.serialization.record.util.IllegalTypeConversionException;
+import org.codehaus.jackson.JsonNode;
+
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.PathNotFoundException;
+import com.jayway.jsonpath.spi.json.JacksonJsonProvider;
+
+public class JsonPathRowRecordReader extends AbstractJsonRowRecordReader {
+    private static final Configuration STRICT_PROVIDER_CONFIGURATION = Configuration.builder().jsonProvider(new JacksonJsonProvider()).build();
+
+    private final LinkedHashMap<String, JsonPath> jsonPaths;
+    private final InputStream in;
+    private RecordSchema schema;
+    private final String dateFormat;
+    private final String timeFormat;
+    private final String timestampFormat;
+
+    public JsonPathRowRecordReader(final LinkedHashMap<String, JsonPath> jsonPaths, final RecordSchema schema, final InputStream in, final ComponentLog logger,
+        final String dateFormat, final String timeFormat, final String timestampFormat)
+        throws MalformedRecordException, IOException {
+        super(in, logger);
+
+        this.dateFormat = dateFormat;
+        this.timeFormat = timeFormat;
+        this.timestampFormat = timestampFormat;
+
+        this.schema = schema;
+        this.jsonPaths = jsonPaths;
+        this.in = in;
+    }
+
+    @Override
+    public void close() throws IOException {
+        in.close();
+    }
+
+    @Override
+    public RecordSchema getSchema() {
+        return schema;
+    }
+
+    @Override
+    protected Record convertJsonNodeToRecord(final JsonNode jsonNode, final RecordSchema schema) throws IOException {
+        if (jsonNode == null) {
+            return null;
+        }
+
+        final DocumentContext ctx = JsonPath.using(STRICT_PROVIDER_CONFIGURATION).parse(jsonNode.toString());
+        final Map<String, Object> values = new HashMap<>(schema.getFieldCount());
+
+        for (final Map.Entry<String, JsonPath> entry : jsonPaths.entrySet()) {
+            final String fieldName = entry.getKey();
+            final DataType desiredType = schema.getDataType(fieldName).orElse(null);
+            if (desiredType == null) {
+                continue;
+            }
+
+            final JsonPath jsonPath = entry.getValue();
+
+            Object value;
+            try {
+                value = ctx.read(jsonPath);
+            } catch (final PathNotFoundException pnfe) {
+                value = null;
+            }
+
+            value = convert(value, desiredType);
```

Comment:
Will add log message; calling the convert() method with null value will simply return null, so all should be good.